crypto/info.c

```diff
@@ -11,9 +11,78 @@
 #include <openssl/crypto.h>
 #include "internal/dso_conf.h"
 #include "e_os.h"
+#include "buildinf.h"
+#include "internal/thread_once.h"
+
+static char *seed_sources = NULL;
+static CRYPTO_ONCE init_info = CRYPTO_ONCE_STATIC_INIT;
+
+DEFINE_RUN_ONCE_STATIC(init_info_strings)
+{
+    {
+        static char seeds[512] = "";
+
+#define add_seeds_string(str)                                           \
```

Comment:
I can't say I'm a fan of macros like this.  Using `strcat` mean performance will be awful, macros aren't going to help.

Reply:
I'm not a fan code copying, which would be the alternative...  and this is the info command, you just have to be kidding re performance, I don't see why that would be an issue here.